File: prolog/evenSum.pl
evenSum([],0).
evenSum([H|T],Sum):-
    0 is mod(H,2),
    evenSum(T,SumT),
    Sum is H + SumT.
evenSum([H|T],Sum):-
    1 is mod(H,2),
    evenSum(T,Sum).

File: haskell/evenSum.hs
evenSum :: Integral i => [i] -> i
-- evenSum [1,2,3,4] = 6
-- evenSum [1,3,5,7] = 0
evenSum [] = 0
evenSum (h:t)
	| (mod h 2) == 0 = h + (evenSum t)
	| otherwise	 = evenSum t
